ebean-test/src/test/java/org/tests/query/sqlquery/SqlQueryTests.java

```diff
@@ -224,6 +241,44 @@ public CustDto map(ResultSet rset, int rowNum) throws SQLException {
 
   private static final CustMapper CUST_MAPPER = new CustMapper();
 
+  @Test
+  void queryUsingTransaction() {
+    ResetBasicData.reset();
+
+    try (Transaction txn = DB.createTransaction()) {
+      String sql = "select id, name, status from o_customer where name is not null";
+      AtomicInteger counter = new AtomicInteger();
+      DB.sqlQuery(sql)
+        .usingTransaction(txn)
+        .mapTo(CUST_MAPPER)
+        .findEach(custDto -> {
+          counter.incrementAndGet();
+          assertThat(custDto.name).isNotNull();
+        });
+
+      assertThat(counter.get()).isGreaterThan(0);
+    }
+  }
+
+  @Test
+  void mapperUsingTransaction() {
+    ResetBasicData.reset();
+
+    try (Transaction txn = DB.createTransaction()) {
+      String sql = "select id, name, status from o_customer where name is not null";
+      AtomicInteger counter = new AtomicInteger();
+      DB.sqlQuery(sql)
+        .mapTo(CUST_MAPPER)
+        .usingTransaction(txn)
+        .findEach(custDto -> {
+          counter.incrementAndGet();
+          assertThat(custDto.name).isNotNull();
```

Comment:
I'm not sure, if a findEach will update Transaction.current(). If so, we can `assertThat(Transaction.current()).isEqualTo(txn)` here